executor/spell_checker.py

```diff
@@ -0,0 +1,82 @@
+__copyright__ = "Copyright (c) 2021 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+import os
+import pickle
+from typing import Dict, Iterable
+
+from jina import DocumentArray, Executor, requests
+from jina.logging.logger import JinaLogger
+from .pyngramspell import PyNgramSpell
+
+cur_dir = os.path.dirname(os.path.abspath(__file__))
+
+
+class SpellChecker(Executor):
+    """A simple spell checker based on BKTree
+
+    It can be trained on your own corpus, on the /train endpoint
+
+    Otherwise it automatically spell corrects your Documents with string contents.
+    The content is overridden.
+    """
+
+    def __init__(
+        self,
+        model_path: str = os.path.join(cur_dir, 'model.pickle'),
+        traversal_paths: Iterable = ['r'],
```

Comment:
I think you should not use Lists as defaults parameters here. Either use a Tuple or use None. The list can lead to bugs because of mutable sequences in python.